toolz/itertoolz.py

```diff
@@ -97,6 +98,32 @@ def groupby(key, seq):
     return rv
 
 
+def indices(*sizes):
+    """ Iterates over a length/shape.
+
+    >>> list(indices(3, 2))
+    [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1)]
+
+    This can help nicely index an array.
+
+    >>> l = [[1, 2],
+    ...      [3, 4],
+    ...      [5, 6]]
+
+    >>> for i, j in indices(3, 2):
+    ...     print("l[%i][%i] = %i" % (i, j, l[i][j]))
+    l[0][0] = 1
+    l[0][1] = 2
+    l[1][0] = 3
+    l[1][1] = 4
+    l[2][0] = 5
+    l[2][1] = 6
```

Comment:
I think I understand, thanks for clarifying! Looking through some of my numpy code I see there are places where I could have used something like this; however, I realized that this is in numpy as `numpy.indices`. I wonder if I would want this when working with normal lists/tuples where numpy was not available. If we are going the route of allowing more functions into toolz but selectivly curating the top level namespace then I would be +1 on adding this, but -0 on putting it in the top level. This is because I think it is not immediatly obvious when this is the right function to use over just standard looping or slice indexing so it is more "advanced" than other functions in toolz.